aws-codeguruprofiler-profilinggroup/src/test/java/software/amazon/codeguruprofiler/profilinggroup/UpdateHandlerTest.java

```diff
@@ -0,0 +1,267 @@
+
+            assertThat(response).isNotNull();
+            assertThat(response.getStatus()).isEqualTo(OperationStatus.SUCCESS);
+            assertThat(response.getCallbackContext()).isNull();
+            assertThat(response.getCallbackDelaySeconds()).isEqualTo(0);
+            assertThat(response.getResourceModel()).isEqualTo(request.getDesiredResourceState());
+            assertThat(response.getMessage()).isNull();
+            assertThat(response.getErrorCode()).isNull();
+        }
+
+        @Test
+        public void testCorrectOperationIsCalled() {
+            subject.handleRequest(proxy, request, null, logger);
+
+            verify(proxy, times(1)).injectCredentialsAndInvokeV2(eq(getPolicyRequest), any());
+            verifyNoMoreInteractions(proxy);
+        }
+
+        @Nested
+        class WhenPolicyWasAttached {
+            @BeforeEach
+            public void setup() {
+                doReturn(
+                    GetPolicyResponse.builder()
+                        .policy("RandomString")
+                        .revisionId(revisionId)
+                        .build()
+                ).when(proxy).injectCredentialsAndInvokeV2(eq(getPolicyRequest), any());
+            }
+
+            @Test
+            public void testCorrectOperationsAreCalled() {
+                subject.handleRequest(proxy, request, null, logger);
+
+                verify(proxy, times(1)).injectCredentialsAndInvokeV2(eq(getPolicyRequest), any());
+                verify(proxy, times(1)).injectCredentialsAndInvokeV2(
+                    eq(RemovePermissionRequest.builder()
+                           .profilingGroupName(profilingGroupName)
+                           .actionGroup(ActionGroup.AGENT_PERMISSIONS)
+                           .revisionId(revisionId)
+                           .build()
+                    ),
+                    any());
+                verifyNoMoreInteractions(proxy);
+            }
+        }
+    }
+
+    @Nested
+    class WhenPermissionsIsProvided {
+        private final PutPermissionRequest putPermissionRequest =
+            PutPermissionRequest.builder()
+                .profilingGroupName(profilingGroupName)
+                .actionGroup(ActionGroup.AGENT_PERMISSIONS)
+                .principals(principals)
+                .build();
+
+        @BeforeEach
+        public void setup() {
+            request =  makeRequest(
+                ResourceModel.builder()
+                    .profilingGroupName(profilingGroupName)
+                    .permissions(
+                        Permissions.builder()
+                            .agentPermissions(
+                                AgentPermissions.builder()
+                                    .principals(principals)
+                                    .build()
+                            )
+                            .build()
+                    )
+                    .build()
+            );
```

Comment:
These levels should soon disappear, I removed one level in [PR #10](https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-codeguru-profiler/pull/10/files).

Reply:
I think I would leave these changes like that for now until you have pushed yours so that I can fix them with rebase/ merge conflict :)

Same comment for the schema change :)